src/ui/components/shared/NewWorkspaceModal/NewWorkspaceModal.tsx

```diff
@@ -271,6 +278,50 @@ function OnboardingModal(props: PropsFromRedux) {
   const [current, setCurrent] = useState<number>(1);
   const [newWorkspace, setNewWorkspace] = useState<Workspace | null>(null);
 
+  const { id: userId } = hooks.useGetUserInfo();
+  const [workspaceName, setWorkspaceName] = useState<string>();
+  const updateDefaultWorkspace = hooks.useUpdateDefaultWorkspace();
+  const createNewWorkspace = hooks.useCreateNewWorkspace(() => {});
+  const inviteNewWorkspaceMember = hooks.useInviteNewWorkspaceMember(() => {});
+  const { activateWorkspaceSubscription } = hooks.useActivateWorkspaceSubscription();
+
+  const handleSelectMembers = async (members: WorkspaceUser[]) => {
+    try {
+      const workspaceResult = await createNewWorkspace({
+        variables: { name: workspaceName, userId },
+      });
+      const workspace = workspaceResult.data.createWorkspace.workspace as Workspace;
+
+      setNewWorkspace(workspace);
+      await updateDefaultWorkspace({
+        variables: {
+          workspaceId: workspace.id,
+        },
+      });
+
+      await Promise.all(
+        members
+          .filter(m => m.email !== "me@me.com")
+          .map(m =>
+            inviteNewWorkspaceMember({ variables: { workspaceId: workspace.id, email: m.email! } })
+          )
+      );
+
+      await activateWorkspaceSubscription({
+        variables: {
+          workspaceId: workspace.id,
+          subscriptionKey: "team-1",
+          seatCount: members.length,
+        },
+      });
+
+      setCurrent(3);
+    } catch (e) {
+      console.error("failed to create workspace and subscription");
+      console.log(e);
+    }
+  };
+
```

Comment:
I lifted these up from the individual screens because I didn't want to create workspaces and invite members if the user bailed before submitting. The result is a lot of service orchestration that is probably better done on the backend.

Reply:
After discussion with @jasonLaster, we decided to pull these back apart and trigger subscription creation with team creation and updating that subscription as members are added and removed.